test/integration/azureDataService.js

```diff
@@ -19,40 +19,164 @@ const azureDataService = new AzureDataService({
-      await azureDataService.uploadIds(timeOfEtl);
+  describe('upload/download functions', () => {
+    after(async function deleteGeneratedFile() {
+      this.timeout(timeout);
+      await azureService.deleteFromAzure(containerName, `${outputFile}-${dateString}-${version}.json`);
+      await azureService.deleteFromAzure(containerName, `${outputFile}-seed-ids-${dateString}.json`);
+      await azureService.deleteFromAzure(containerName, `${outputFile}-summary-${dateString}-${version}.json`);
     });
-  });
 
-  describe('download functions', () => {
-    it('should get latest data', async () => {
+    it('should upload and download latest data', async () => {
+      await azureDataService.uploadData(timeOfEtl);
       const { data, date } = await azureDataService.getLatestData();
       expect(data).to.exist;
       expect(date).to.exist;
     });
 
-    it('should get latest ids', async () => {
+    it('should upload and download latest ids', async () => {
+      await azureDataService.uploadIds(timeOfEtl);
       const { data, date } = await azureDataService.getLatestIds();
       expect(data).to.exist;
       expect(date).to.exist;
     });
   });
+
+  function uploadOldDateStampedData(date, days) {
+    return azureService.uploadToAzure(
+      containerName,
+      azureDataService.localFile,
+      `${azureDataService.outputFile}${azureDataService.getSuffixWithVersion(moment(date).subtract(days, 'days'))}`
+    );
+  }
+  function uploadOldDateStampedSummary(date, days) {
+    return azureService.uploadToAzure(
+      containerName,
+      azureDataService.localFile,
+      `${azureDataService.outputFile}-${azureDataService.summaryFile}${azureDataService.getSuffixWithVersion(moment(date).subtract(days, 'days'))}`
+    );
+  }
+
+  function uploadOldDateStampedIdList(date, days) {
+    return azureService.uploadToAzure(
+      containerName,
+      azureDataService.localSeedIdFile,
+      `${azureDataService.seedIdFile}${azureDataService.getSuffix(moment(date).subtract(days, 'days'))}`
+    );
+  }
+
+  describe('prune functions', () => {
+    after(async function deleteGeneratedFile() {
+      this.timeout(timeout);
+      await azureService.deleteFromAzure(containerName, `${azureDataService.outputFile}-20180321-${version}.json`);
+      await azureService.deleteFromAzure(containerName, `${azureDataService.outputFile}-${azureDataService.summaryFile}-20180321-${version}.json`);
+      await azureService.deleteFromAzure(containerName, `${azureDataService.seedIdFile}-20180321.json`);
+      await azureService.deleteFromAzure(containerName, `${azureDataService.outputFile}-20180314-${version}.json`);
+      await azureService.deleteFromAzure(containerName, `${azureDataService.outputFile}-${azureDataService.summaryFile}-20180314-${version}.json`);
+      await azureService.deleteFromAzure(containerName, `${azureDataService.seedIdFile}-20180314.json`);
+    });
+
+    it('should remove data files older than date', async () => {
```

Comment:
Given this test and the 2 following it execute the same function i.e `pruneFilesOlderThan()` and use some of the same functions to get the list of files in the blob. Could the setup be done for all 3 tests in one go and have the filters and expectations run separately? I know it is largely personal preference, the reason for changing would be to make it clear there is just one function being run that ends up prune all files.
It would also reduce the execution time.

Reply:
I think that makes sense, as that way there's a symmetry to the setup before and the teardown afterwards.